include/ranges.h

```diff
@@ -278,8 +277,6 @@ class range : public detail::coordinate<range<Dims>, Dims> {
 	}
 
   private:
-	friend class detail::coordinate<range<Dims>, Dims>;
```

Comment:
Why do we no longer need this friend decl?

Reply:
It was used to allow the base class to default-construct `range` for every dimensionality, but the constructor is now public. The thing you actually noticed was probably that I forgot to remove the private default constructor (now fixed).